### ISSUE TITLE
Momentary auto resize on refocus

### ISSUE THREAD

**Issue by andreiosg** [pop-os/shell#1330]
**(1) Issue/Bug Description:**
Gnome apps resize when refocused. Attached a demo gif to show it in action.
![pop_shell_issue](https://user-images.githubusercontent.com/35179329/149939776-747e92c6-50a6-4c43-a85e-9a70596ff7f7.gif)

Doesn't seem to affect Alacritty, Firefox or OBS.

**(2) Steps to reproduce (if you know):**
Tested on Gnome 41.3 Wayland, when gnome apps are refocused they automatically change window size momentarily.

**(3) Expected behavior:**
The window shouldn't momentarily change size on refocus.

**(4) Distribution (run `cat /etc/os-release`):**
 NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://bugs.archlinux.org/"
LOGO=archlinux-logo

**(5) Gnome Shell version:**
GNOME Shell 41.3

**(6) Pop Shell version (run `apt policy pop-shell` or provide the latest commit if building locally):**
```
> paru -Qe | grep pop-shell
gnome-shell-extension-pop-shell-git r847.21745c4-1
``` 

**(7) Where was Pop Shell installed from:**
Installed from the AUR: 
https://aur.archlinux.org/packages/gnome-shell-extension-pop-shell-git/

**(8) Monitor Setup (2 x 1080p, 4K, Primary(Horizontal), Secondary(Vertical), etc):**
2 x 4K Primary, FHD Secondary

**(9) Other Installed/Enabled Extensions:**
Vitals, AppIndicator and KStatusNotifierItem Support, Application Volume Mixer, Removable Drive Menu, Workspaces Bar.
They don't seem to affect the issue. Encountered the same issue with all of them disabled.

**(10) Other Notes:**
Seems to have started a few versions ago.

**Comment by ehardesty** [comment]
I have the same issue in Fedora with Mutter 41.3. Reverting Mutter to versions prior to 41.3 fixes it.

The real issue is in Mutter/GTK: https://gitlab.gnome.org/GNOME/mutter/-/issues/2091

**Comment by dalingrin** [comment]
I have the same issue in Fedora with Mutter 41.3. Reverting Mutter to versions prior to 41.3 fixes it.

The real issue is in Mutter/GTK: https://gitlab.gnome.org/GNOME/mutter/-/issues/2091

**Comment by andreiosg** [comment]
Thanks for the tip! Should I leave the issue until it's fixed in Mutter so others can find it?

**Comment by nikolowry** [comment]
> Thanks for the tip! Should I leave the issue until it's fixed in Mutter so others can find it?

I've been going crazy trying to figure this out. Very happy it was visible and reported here.

**Comment by dalingrin** [comment]
For anyone using Fedora, I made a COPR for Mutter 41.3 that includes the fix from Sebastian Keller

https://copr.fedorainfracloud.org/coprs/dalingrin/Pop_Shell/

**Comment by ejawhary** [comment]
How can I implement this in arch? I tried downgrading mutter and gnome-shell using pacman -U however after I reboot gnome shell wont start. It says there been an unrecoverable error and to contact the system admin. I managed to get to the cli and update mutter and gnome-shell to the latest version and i can boot into gnome again.
any ideas?

**Comment by ejawhary** [comment]
@ehardesty Thanks for the fast reply and links. I found out that I can just switch to Xorg to solve the issue. I will revert back to Wayland when it is officially fixed. Thanks for your help.

**Comment by andreiosg** [comment]
Seems to be fixed on Arch with `mutter 41.4`. ^^
